rust/src/main.rs

```diff
@@ -128,9 +135,9 @@ fn gen(args: Args) -> Result<()> {
     let mut sizes = BufWriter::new(File::create(format!("{}.sizes", args.output))?);
     let mut trecids = BufWriter::new(File::create(format!("{}.documents", args.output))?);
 
-    let bar = ProgressBar::new(num_documents as u64);
-    bar.set_style(pb_style());
-    bar.set_draw_delta(num_documents as u64 / 100);
+    let progress = ProgressBar::new(u64::from(num_documents));
```

Comment:
Here, I'm using `from` instead of `as` for robustness really, if somehow `num_documents` changes type for whatever reason, `as` might behave shady, whereas `from` won't accept anything that's not kosher.